torch/_dynamo/guards.py

```diff
@@ -413,23 +413,33 @@ def TENSOR_MATCH(self, guard: Guard):
             value = self.get(guard.name)
             assert isinstance(value, torch.Tensor)
             tensor_name = self.arg_ref(guard)
-            self.tensor_check_names.append(tensor_name)
-            self.tensor_check_examples.append(value)
-
-            # STOP - DO NOT USE id_ref FOR TENSORS - TENSOR INVALIDATION RULES DIFFER
-            self.tensor_check_ids[tensor_name] = id(value)
-
-            # Note: Guard code produced for tensor_match is a little different.
-            # We accumulate tensor names, then do a single install of `___check_tensors`.
-            # See _guards.cpp and TensorGuard for more information.
-            # TODO(voz): Add tensor matching code to export
-            # Note: this is a bit of a special case, and so does not use _produce_guard_code
-            guard.set_export_info(
-                "TENSOR_MATCH",
-                weakref.ref(type(value)),
-                None,
-                weakref.ref(value),
-            )
+            # [Note - On Export Tensor Guards]
+            #
+            # In eager mode, tensor guards are evaluated through C++, in guards.cpp
+            # see [Note - On Eager Tensor Guards] for more info.
+            #
+            # In export mode, we instead maintain parallel logic between C++ and python
+            # here, with an exception of checking the dispatch key - with the idea that a dispatch key
+            # is an entirely runtime notion that would make no sense to keep in an exported graph.
+            #
+            # The list of tensor fields and calls we care about can be found in `terms` below.
+            if self.check_fn_manager.output_graph.export:
+                # An interesting observation, is that the code below
+                # + a dispatch_key check + TYPE_MATCH performs almost identically
+                # to our C++ guards on a small sweep.
```

Comment:
I'm finding this hard to square with the fact that we intentionally don't install guards on all module parameters because it is too expensive. If it really is this cheap, then it should also be cheap to properly detect when module parameters have gotten invalidated...

Reply:
I can remove this comment :)